platform/ios/src/MGLMapView.mm

```diff
@@ -2698,17 +2755,29 @@ - (MGLAnnotationTag)annotationTagAtPoint:(CGPoint)point persistingResults:(BOOL)
             id <MGLAnnotation> annotation = [self annotationWithTag:annotationTag];
             NSAssert(annotation, @"Unknown annotation found nearby tap");
             
-            MGLAnnotationImage *annotationImage = [self imageOfAnnotationWithTag:annotationTag];
-            if ( ! annotationImage.enabled)
+            MGLAnnotationContext annotationContext = _annotationContextsByAnnotationTag[annotationTag];
+           
+            CGRect annotationRect;
+            if (annotationContext.annotationView)
+            {
+                annotationRect = annotationContext.annotationView.frame;
```

Comment:
Real views can do their own hit-testing. We [let the user dot do its own hit testing](https://github.com/mapbox/mapbox-gl-native/blob/a4b3ff55403e648e86a8539503bf2732454a242c/platform/ios/src/MGLMapView.mm#L1294) because it’s round. An [irregularly shaped annotation view](https://www.mapbox.com/ios-sdk/examples/marker-image/) would also benefit from this treatment.